app/src/main/java/in/co/rajkumaar/amritarepo/activities/LaunchingActivity.java

```diff
@@ -576,19 +529,133 @@ private void intentSemActivity(int position, String title) {
     @Override
     public void onStart() {
         super.onStart();
-        active = true;
+        // Create a listener to track request state updates.
+        listener = state -> {
+            // (Optional) Provide a download progress bar.
+            if (state.installStatus() == InstallStatus.DOWNLOADED) {
+                popupSnackbarForCompleteUpdate();
+            }
+        };
+        // Before starting an update, register a listener for updates.
+        appUpdateManager.registerListener(listener);
+        appUpdateInfoTask.addOnSuccessListener(appUpdateInfo -> {
+            if (appUpdateInfo.updateAvailability() == UpdateAvailability.UPDATE_AVAILABLE){
+                if(appUpdateInfo.updatePriority() > 3){
+                    try {
+                        appUpdateManager.startUpdateFlowForResult(
+                                // Pass the intent that is returned by 'getAppUpdateInfo()'.
+                                appUpdateInfo,
+                                // Or 'AppUpdateType.FLEXIBLE' for flexible updates.
+                                AppUpdateType.IMMEDIATE,
```

Comment:
Please remove the if else (for priority) and use conditional operators so that Lines of Code gets reduced. Another important thing is that I can't control this priority value from Developer Console. Hence, please make the update IMMEDIATE if the update is stalled for more than a day. 

Pseudo code : `stalledForMoreThanADay ? IMMEDIATE : FLEXIBLE`
Ref : https://developer.android.com/guide/playcore/in-app-updates#check-staleness

Reply:
value for DAYS_FOR_FLEXIBLE_UPDATE && DAYS_FOR_IMMEDIATE_UPDATE?

Reply:
```java
if (appUpdateInfo.updateAvailability() == UpdateAvailability.UPDATE_AVAILABLE) {
        int stalledForMoreThanADay = 
           appUpdateInfo.clientVersionStalenessDays() != null ? appUpdateInfo.clientVersionStalenessDays() > 1 : true;
        // Then set the update type like (stalledForMoreThanADay ? IMMEDIATE : FLEXIBLE)
}